lib/widgets/atoms/explore_filter.dart

```diff
@@ -79,6 +81,8 @@ class ExploreFilter extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     final AppLocalizations l10n = AppLocalizations.of(context)!;
+    bool userFiltersSelected =
+        (skills!.isNotEmpty || languages!.isNotEmpty || countries!.isNotEmpty);
```

Comment:
You need to handle the case when the lists are null or else the app will crash right here. You can either make the lists a required field (caller passes an empty list if no filters are applied), or keep them as nullable (preferred) and do null checking on the lists before using them. 

There are other operators besides the dangerous null-assertion operator (!): you could use the conditional access operator (?), the null-coalescing operator (??), or plain old 'if' operators to check equality against 'null'.

This codelab has some useful tips: https://dart.dev/codelabs/null-safety

Reply:
Hi Miguel, I went through the codelab on null safety. It's a great resource. Thanks for sharing. 
Yeah, I agree that it is preferred to keep the lists nullable, in the scenario there are no filters applied. I have added some more logic that takes away the need for the null-assertion using some if statements. Please take a look.